server/src/main/java/org/opensearch/index/shard/IndexShard.java

```diff
@@ -1374,18 +1373,31 @@ public GatedCloseable<IndexCommit> acquireSafeIndexCommit() throws EngineExcepti
     }
 
     /**
-     * Returns the lastest Replication Checkpoint that shard received
+     * Returns the IndexShardSate of current shard
      */
-    public ReplicationCheckpoint getLatestReplicationCheckpoint() {
-        return new ReplicationCheckpoint(shardId, 0, 0, 0, 0);
+    public IndexShardState getState() {
```

Comment:
I think this method is unnecessary we already have `indexShard.state()`?

Reply:
Got it. Will remove this in next commit